vms/platformvm/txs/executor/staker_tx_verification.go

```diff
@@ -468,7 +468,7 @@ func verifyAddPermissionlessValidatorTx(
 
 	validatorRules, err := getValidatorRules(backend, chainState, tx.Subnet)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed retrieving validator rules: %w", err)
```

Comment:
nit just to make errors more easily traceable. The alternative is the everpresent `database.ErrNotFound `which does not help error localization